packages/walletconnect/src/index.ts

```diff
@@ -125,15 +125,13 @@ export class WalletConnect extends Connector {
       await this.isomorphicInitialize()
       if (!this.provider?.connected) throw Error('No existing connection')
 
-      // for walletconnect, we always use sequential instead of parallel fetches because otherwise
-      // chainId defaults to 1 even if the connecting wallet isn't on mainnet
-      const accounts = await this.provider?.request<string[]>({ method: 'eth_accounts' })
+      // Wallets may resolve eth_chainId and hang on eth_accounts pending user interaction, which may include changing
+      // chains; they should be requested serially, with accounts first, so that the chainId can settle.
+      const accounts = await this.provider.request<string[]>({ method: 'eth_accounts' })
+      const chainId = await this.provider.request<string>({ method: 'eth_chainId' })
       if (!accounts.length) throw new Error('No accounts returned')
-      const chainId = await this.provider
-        .request<string | number>({ method: 'eth_chainId' })
-        .then((chainId) => parseChainId(chainId))
 
-      this.actions.update({ chainId, accounts })
+      this.actions.update({ chainId: parseChainId(chainId), accounts })
```

Comment:
nit: maybe `chainId` should already be parsed? `parseChainId(await ....)`, otherwise I think this line is a bit confusing: `chainId: parseChainId(chainId)` vs `this.actions.update({ chainId })`

Reply:
If considered a worthy nit, this most likely applies to other files too.

Reply:
fwiw, I agree with you here, but I don't want to change it because it expands the scope of this PR - I've omitted it to de-risk this bug fix.